chore(web-analytics): move the demo data pre-aggregation backfill out of mprocs

lricoy opened the pull request:
## Problem

We can keep all the logic in Dagster and make it simpler.

Before we were triggering it on the `generate_demo_data.py` but it was for v1 only. Personally, I think this is cleaner. The only downside is that it may take a bit longer to run the first time as the scheduler checks in 10 minutes intervals if it should backfill again, but the benefit there is that it will fix stale data as well. We can always trigger it from Dagster UI if we need.

A future improvement here is that we can use that approach alongside the data accuracy job to rerun some partitions. 

## Changes

Move the logic to automatically pre-aggregate local data to live in Dagster instead of mprocs.

## How did you test this code?

Killing all my containers and get a new environment up. 

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Optional, but helpful for our content team! -->
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->

lricoy committed:
chore: move the demo data pre-aggregation backfill logic into a sensor

lricoy committed:
chore: prefer a simple scheduler

lricoy committed:
chore: remove manual dagster materialization from generate_demo_data

Automated schedule now handles web analytics pre-aggregation

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

jabahamondes commented on dags/web_preaggregated.py:
```diff
@@ -215,3 +216,87 @@ def web_pre_aggregate_current_day_schedule(context: dagster.ScheduleEvaluationCo
+    if TEST:
+        return
+
+    try:
```
Can this be tested locally? or we are confident it's fine?

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

lricoy committed:
fix(dagster): use get_run_stats to access DagsterRun end_time

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev

lricoy committed:
fix: conditionally register backfill schedule in TEST mode

The web_analytics_v2_backfill_schedule was causing CI failures in test
environments because it accesses Dagster instance methods that may not
be initialized during test runs. This change conditionally excludes the
schedule from the Definitions when TEST=True, preventing the schedule
from being registered during test execution.

lricoy committed:
Merge branch 'master' into chore/pre-aggregated-tables-for-dev
